src/Build/Evaluation/Evaluator.cs

```diff
@@ -1348,13 +1348,30 @@ private void LogPropertyReassignment(P predecessor, P property, string location)
 
             if (newValue != oldValue)
             {
-                _evaluationLoggingContext.LogComment(
-                    MessageImportance.Low,
-                    "PropertyReassignment",
-                    property.Name,
-                    newValue,
```

Comment:
OMG if I'm reading this right, I think the newValue and the oldValue were swapped in the old code! And it's probably my fault! How did we not notice this all these years!

Reply:
![image](https://github.com/dotnet/msbuild/assets/679326/d705ce59-c0f1-4fc0-8670-4225d691ff0a)

Good news is that the new code fixes this, old on the right, new on the left.

Reply:
ah, no, the old code is correct, the new value comes before the previous value. It's a recent regression in the viewer:
https://github.com/KirillOsenkov/MSBuildStructuredLog/commit/89065d2e2a8159a478afab788a5a0ead6f94e91d

It is still my fault, however fortunately this only broke a few weeks back. Everything in MSBuild is correct!